appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/NewYoungAndroidProjectWizard.java

```diff
@@ -55,8 +55,10 @@ public NewYoungAndroidProjectWizard(final ProjectToolbar toolbar) {
     projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel(), new Validator() {
       @Override
       public boolean validate(String value) {
+        
+        value = value.trim();
         errorMessage = TextValidators.getErrorMessage(value);
-        if (errorMessage.length()>0){
+        if (errorMessage.equals(MESSAGES.firstCharProjectNameError()) || errorMessage.equals(MESSAGES.invalidCharProjectNameError())){ // !errorMessage == MESSAGES.whitespaceProjectNameError();
```

Comment:
See my note in LabeledTextBox about adding a warning mechanism. If we were to add other errors in the future we'd need to account for them here. By separating the concerns and having a warning mechanism in addition to the error mechanism, we can avoid future errors as a problem to be addressed here.